src/smc-webapp/custom-software/init.ts

```diff
@@ -0,0 +1,164 @@
+// Manage DB <-> UI integration of available *custom* compute images
+// TODO: also get rid of hardcoded legacy software images
+
+const { redux, Store, Actions, Table } = require("../app-framework");
+import { Map as iMap } from "immutable";
+import { CUSTOM_IMG_PREFIX } from "./util";
+import { join as path_join } from "path";
+const { capitalize } = require("smc-util/misc");
+
+export const NAME = "compute_images";
+
+// this must match db-schema.compute_images → field type → allowed values
+// legacy image names are "default", "exp", or a timestamp-string
+// custom iamges are "custom/<image-id>/<tag, usually latest>"
+// the "custom/" string is supposed to be CUSTOM_IMG_PREFIX!
+export type ComputeImageTypes = "legacy" | "custom";
+
+// this must be compatible with db-schema.compute_images → field keys
+export type ComputeImageKeys =
+  | "id"
+  | "src"
+  | "type"
+  | "display"
+  | "url"
+  | "desc"
+  | "path"
+  | "search_str"
+  | "display_tag"
+  | "disabled";
+
+export type ComputeImage = iMap<ComputeImageKeys, string>;
```

Comment:
It doesn't seem to make tooo much a difference here but we have a type for this usage:

```
import { TypedMap } from "../app-framework/TypedMap";

ComputeImage = TypedMap<{
  id: string,
  src: string,
  // ... etc
}
```

Then you can specify some of them as optional. As written in your code now, they're all optional.
Also allows you to do stuff like 
```
keyof ComputeImage
```